recipes/out_ptr/all/conanfile.py

```diff
@@ -0,0 +1,32 @@
+import os
+
+from conan import ConanFile
+from conan.tools.files import get
+
+required_conan_version = ">=1.45.0"
+
+class OutPtrConan(ConanFile):
+    name = "out_ptr"
+    version = "ea379b2"
+
+    # Optional metadata
+    license = "Apache-2.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/soasis/out_ptr"
+    description = "a C++11 implementation of std::out_ptr (p1132), as a standalone library"
+    topics = ("utility", "backport")
+    generators = "CMakeToolchain"
```

Comment:
```suggestion
    generators = "CMakeToolchain"
    settings = "os", "arch", "build_type", "compiler"
```
Use full settings always. It will be the default behavior for Conan 2.0. You still can manage the package ID as you have done